native/swift/Example/Example/UI/LoginView.swift

```diff
@@ -69,16 +69,17 @@ struct LoginView: View {
                     callbackUrl: "x-wordpress-app://login-callback"
                 )
 
-                let loginClient = WordPressLoginClient(urlSession: .shared)
-                let loginUrl = try await loginClient.loginURL(forSite: url, application: application)
+                let loginClient = WpLoginClient(urlSession: .shared)
+                let details = try await loginClient.apiDiscovery(siteUrl: url)
+                let loginUrl = try await details.loginURL(for: application)
 
                 let callbackUrl = try await self.webAuthenticationSession.authenticate(
                     using: loginUrl,
                     callbackURLScheme: "x-wordpress-app"
                 )
 
                 let loginDetails = try loginClient.credentials(from: callbackUrl)
-                try await loginManager.setLoginCredentials(to: loginDetails)
+                try await loginManager.setLoginCredentials(to: loginDetails, apiRootURL: details.apiRootUrl.asURL())
```

Comment:
The changes in this file are how login should work: the client needs both the login URL and the API root URL.